Source/ACE.Server/Entity/Actions/ActionQueue.cs

```diff
@@ -30,5 +30,7 @@ public void EnqueueAction(IAction action)
         {
             Queue.Enqueue(action);
         }
+
+        public int Count { get { return Queue.Count; } }
```

Comment:
This is suspect. Getting the queue count is a very costly operation.

Alternatively, you can use IsEmpty.